x/tokenomics/module/abci.go

```diff
@@ -10,29 +10,49 @@ import (
 )
 
 // EndBlocker called at every block and settles all pending claims.
-func EndBlocker(ctx sdk.Context, k keeper.Keeper) error {
+func EndBlocker(ctx sdk.Context, k keeper.Keeper) (err error) {
 	logger := k.Logger().With("method", "EndBlocker")
 	// NB: There are two main reasons why we settle expiring claims in the end
 	// instead of when a proof is submitted:
 	// 1. Logic - Probabilistic proof allows claims to be settled (i.e. rewarded)
 	//    even without a proof to be able to scale to unbounded Claims & Proofs.
 	// 2. Implementation - This cannot be done from the `x/proof` module because
 	//    it would create a circular dependency.
-	numClaimsSettled, numClaimsExpired, relaysPerServiceMap, err := k.SettlePendingClaims(ctx)
+	numClaimsSettled,
+		numClaimsExpired,
+		relaysPerServiceMap,
+		computeUnitsPerServiceMap,
+		err := k.SettlePendingClaims(ctx)
 	if err != nil {
 		logger.Error(fmt.Sprintf("could not settle pending claims due to error %v", err))
 		return err
 	}
 
-	defer telemetry.ClaimCounter(
-		telemetry.ClaimProofStageSettled,
-		func() uint64 { return numClaimsSettled },
-	)
+	// Accumulate compute units for metrics.
+	// TODO_IMPROVE(@bryanchriswhite, @red-0ne): It would be preferable to have telemetry
+	// counter functions return an "event" or "event set", similar to how polylog/zerolog work.
+	var numComputeUnits uint64
+	for _, serviceComputeUnits := range computeUnitsPerServiceMap {
+		numComputeUnits += serviceComputeUnits
+	}
 
-	defer telemetry.ClaimCounter(
-		telemetry.ClaimProofStageExpired,
-		func() uint64 { return numClaimsExpired },
-	)
+	defer func() {
```

Comment:
```
// Package telemetry provides a set of functions for incrementing counters which track
// various events across the codebase. Typically, calls to these counter functions SHOULD
// be made inside deferred anonymous functions so that they will reference the final values
// of their inputs. Any instrumented piece of code which contains branching logic with respect
// its counter function inputs is subject to this constraint (i.e. MUST defer).
```